README.md

```diff
@@ -0,0 +1,526 @@
+### Device Profile Sample
+
+You should change all `valueType`, `NodeName` and `name` to your real `devices/device.yml`.
+
+- `NodeName` should be use `DB4.DBX0.0` format
+
+```yaml
+nname: S7-Device
+manufacturer: YIQISOFT
+description: Example of S7 Device
+model: Siemens S7
+labels: [ISO-on-TCP]
+deviceResources:
+  - name: bool
+    description: PLC bool
+    isHidden: false
+    properties:
+      valueType: Bool
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBX0.0
+  - name: byte
+    description: PLC byte
+    isHidden: false
+    properties:
+      valueType: Uint8
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBB1
+  - name: word
+    description: PLC word
+    isHidden: false
+    properties:
+      valueType: Int16
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBW2
+  - name: dword
+    description: PLC dword
+    isHidden: false
+    properties:
+      valueType: Int32
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBD4
+  - name: int
+    description: PLC int
+    isHidden: false
+    properties:
+      valueType: Int16
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBW8
+  - name: dint
+    description: PLC dint
+    isHidden: false
+    properties:
+      valueType: Int32
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBW10
+  - name: real
+    description: PLC real
+    isHidden: false
+    properties:
+      valueType: Float32
+      readWrite: RW
+    attributes:
+      NodeName: DB4.DBD14
+  - name: heartbeat
+    description: PLC heartbeat
+    isHidden: false
+    properties:
+      valueType: Int16
+      readWrite: RW
+    attributes:
+      NodeName: DB1.DBW160
+deviceCommands:
+  - name: AllResource
+    isHidden: false
+    readWrite: RW
+    resourceOperations:
+      - deviceResource: bool
+        defaultValue: 'false'
+      - deviceResource: byte
+        defaultValue: '0'
+      - deviceResource: word
+        defaultValue: '0'
+      - deviceResource: dword
+        defaultValue: '0'
+      - deviceResource: int
+        defaultValue: '0'
+      - deviceResource: dint
+        defaultValue: '0'
+      - deviceResource: real
+        defaultValue: '0'
+      - deviceResource: heartbeat
+        defaultValue: '0'
+```
+
+### Device Sample
+
+Change `Host`, `Port`, `Rack`, `Slot`, `Timeout`, `IdleTimeout`, `interval` and others(if you want) to your real `Configuration`.
+
+#### S7 protocol in EdgeX device:
+
+| Key         | Default value | Note                             |
+| ----------- | ------------- | -------------------------------- |
+| Host        | N/A           | S7 ip address                    |
+| Port        | N/A           | S7 port, e.g. 102                |
+| Rack        | 0             | Rack number                      |
+| Slot        | 1             | Slot number                      |
+| Timeout     | 30            | connect to S7 timeout, seconds   |
+| IdleTimeout | 30            | connection idle timeout, seconds |
```

Comment:
Are theses truly defaulted? Current validation returns error is they are missing.